modules/reindex/src/main/java/org/opensearch/index/reindex/ReindexValidator.java

```diff
@@ -45,35 +40,28 @@
 import org.opensearch.cluster.ClusterState;
 import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
 import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.Strings;
 import org.opensearch.common.logging.DeprecationLogger;
-import org.opensearch.common.regex.Regex;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.index.IndexNotFoundException;
 import org.opensearch.search.builder.SearchSourceBuilder;
 
-import java.util.List;
-
 class ReindexValidator {
     private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(ReindexValidator.class);
     static final String SORT_DEPRECATED_MESSAGE = "The sort option in reindex is deprecated. " +
         "Instead consider using query filtering to find the desired subset of data.";
 
-    private final CharacterRunAutomaton remoteWhitelist;
     private final ClusterService clusterService;
     private final IndexNameExpressionResolver resolver;
     private final AutoCreateIndex autoCreateIndex;
 
     ReindexValidator(Settings settings, ClusterService clusterService, IndexNameExpressionResolver resolver,
                      AutoCreateIndex autoCreateIndex) {
-        this.remoteWhitelist = buildRemoteWhitelist(TransportReindexAction.REMOTE_CLUSTER_WHITELIST.get(settings));
         this.clusterService = clusterService;
         this.resolver = resolver;
         this.autoCreateIndex = autoCreateIndex;
     }
 
     void initialValidation(ReindexRequest request) {
-        checkRemoteWhitelist(remoteWhitelist, request.getRemoteInfo());
```

Comment:
Wondering if instead of removing the support of whitelist completely, should we move the logic behind a new Setting which allows checks to be performed optionally. We can even evaluate and make it dynamic based on need. This will ensure IP whitelisting as a feature is still supported and can still be used by community based on need.

Reply:
Replied about the keeping the whitelist feature in the previous comment.